server/schedulers/utils.go

```diff
@@ -153,10 +153,20 @@ func minBalanceDiff(count uint64) float64 {
 	return math.Sqrt(float64(count))
 }
 
+func takeInfluence(store *core.StoreInfo, storeDiff *schedule.StoreDiff) {
+	store.LeaderCount += storeDiff.LeaderCount
+	store.RegionCount += storeDiff.RegionCount
+	store.LeaderSize += int64(storeDiff.LeaderSize)
+	store.RegionSize += int64(storeDiff.RegionSize)
+}
+
 // shouldBalance returns true if we should balance the source and target store.
 // The min balance diff provides a buffer to make the cluster stable, so that we
 // don't need to schedule very frequently.
-func shouldBalance(source, target *core.StoreInfo, kind core.ResourceKind) bool {
+func shouldBalance(source, target *core.StoreInfo, kind core.ResourceKind, opInfluence schedule.DiffMap) bool {
+	takeInfluence(source, opInfluence.GetStoreDiff(source.GetId()))
+	takeInfluence(target, opInfluence.GetStoreDiff(target.GetId()))
```

Comment:
I think we need a test to verify `shouldBalance` returns false if there are many pending operators on the store.